pkg/scd/operational_intents_handler.go

```diff
@@ -490,6 +500,17 @@ func (a *Server) upsertOperationalIntentReference(ctx context.Context, authorize
 					}
 				}
 
+				// If an implicit subscription creation is requested, we need to check if:
+				// - a previous subscription was attached,
+				// - if so, if it was an implicit subscription
+				// - if so, if we can remove it after creating the new implicit subscription
+				removePreviousImplicitSubscription, err = subscriptionCanBeRemoved(ctx, r, previousSubscriptionID)
+				if err != nil {
+					return stacktrace.Propagate(err, "Could not determine if previous Subscription can be removed")
+				}
+
+				// We _could_ reuse the previous subscription ID if it was an implicit subscription,
```

Comment:
The comment does not entirely reflect my intention, I will reword it.

in short: when an update is done and the parameters for the creation of an implicit subscription are passed, my understanding is that we actually _want_ to create a new implicit subscription.

Clients are allowed to change the subscription attached to an OIR, and from my understanding, I think it is valid to give the option to replace an existing subscription with an implicit one.

Clients that wish to reuse an existing subscription (whether or not it is an implicit one) should specify the existing subscription ID in the call parameters.

Reply:
To be more precise: there was no design discussion about this – this is my interpretation of how the API should work.

If you think this needs discussing/clarifying we can either open an issue, discuss here or have a chat at the next weekly meeting.

Reply:
> when an update is done and the parameters for the creation of an implicit subscription are passed, my understanding is that we actually want to create a new implicit subscription.

Yes, when [`new_subscription`](https://github.com/astm-utm/Protocol/blob/d26d3222b9816a5bc90d00287c9746c4255f8f0d/utm.yaml#L766) is specified, a new implicit subscription [should be created](https://github.com/astm-utm/Protocol/blob/d26d3222b9816a5bc90d00287c9746c4255f8f0d/utm.yaml#L771).

> I think it is valid to give the option to replace an existing subscription with an implicit one.

I don't know of an actual use case for this, but I agree this is what should happen if someone specifies `new_subscription` on an op intent with an existing subscription.  And, if that existing subscription was implicit and no longer has any dependent operational intents, then I would expect the existing subscription to be automatically deleted by the DSS since it was implicit (managed by DSS).

> Clients that wish to reuse an existing subscription (whether or not it is an implicit one) should specify the existing subscription ID in the call parameters.

Agreed.

> If you think this needs discussing/clarifying we can either open an issue, discuss here or have a chat at the next weekly meeting.

I don't think there is any ambiguity here so I don't think there's need for further discussion.  However, if someone thinks otherwise then we can certainly have that discussion.